scarb/src/bin/scarb/args.rs

```diff
@@ -292,6 +304,46 @@ pub struct GitRefGroup {
     pub rev: Option<String>,
 }
 
+/// Profile specifier.
+#[derive(Parser, Clone, Debug)]
+pub struct ProfileSpec {
+    /// Specify profile to use by name.
+    #[arg(short = 'P', long, value_name = "SPEC")]
+    pub profile: Option<SmolStr>,
+    /// Use release profile.
+    #[arg(long, hide_short_help = true)]
+    pub release: bool,
+    /// Use dev profile.
+    #[arg(long, hide_short_help = true)]
+    pub dev: bool,
+}
+
+impl ProfileSpec {
+    fn verify(&self) -> Result<()> {
+        if self.dev && self.release {
+            bail!("cannot specify both `--dev` and `--release`");
+        }
+        if self.release && self.profile.is_some() {
+            bail!("cannot specify both `--release` and `--profile`");
+        }
+        if self.release && self.dev {
+            bail!("cannot specify both `--release` and `--dev`");
+        }
+        Ok(())
+    }
```

Comment:
Clap has this built-in: https://github.com/software-mansion/scarb/blob/e10047e932fb2c53bceca01d87ba60f9dd240775/scarb/src/bin/scarb/args.rs#L281